### PR TITLE
tr_Revamp_active_stations_and_detector_status_226

### LINKED ISSUE #226: Revamp active_stations and detector_status models once VDS metadata lands
The `int_clearinghouse__active_stations` and `int_diagnostics__detector_status` models are key models for determining which stations are deployed and considered reliable on any given day. Unfortunately, we have a few concerns about the data quality around for detectors (cf. some discussion in #217 ).

Once we have the VDS metadata from #115, we should revamp those models with the following constraints:

1. No gaps in dates, no null values
2. No gaps in stations, no null values
3. No gaps in detectors/lanes, no null values
4. No rows detectors/lanes that don't exist (i.e., the eighth lane in a highway that only has two lanes)

Once this is done, the `int_clearinghouse__real_detectors` model in #209 can be removed.

### PR DESCRIPTION
**Pull request by tnrahim**
This pull request will update int_clearinghouse__active_stations and int_diagnostics__detector_status models  with VDS metadata to have the following constraints.
 It will close #226.

No gaps in dates, no null values
No gaps in stations, no null values
No gaps in detectors/lanes, no null values
No rows detectors/lanes that don't exist (i.e., the eighth lane in a highway that only has two lanes)

### CODE REVIEW

**Comment by tnrahim** [comment]
Added Date_spine logic to int_diagnostics__detector_status

**Comment by kengodleskidot** [comment]
Very nice work @tnrahim. I have some updated detector status logic I added to my PR #332 if you have a chance to review it. When trying to do something similar I ran into an issue with the counts associated with the detector statuses not looking right.

**Comment by tnrahim** [comment]
Will close #310 as well.

**Comment by tnrahim** [comment]
> @tnrahim Can you take a look at what @kengodleskidot is doing in #332? You both are driving at similar things, and I want to make sure you're not duplicating too much effort.

Sounds good. I've scheduled time tomorrow to connect on this.

**Comment by tnrahim** [comment]
I'm getting the error 

 'Database Error in model imputation__detector_summary (models/marts/imputation/imputation__detector_summary.sql)
  002003 (02000): SQL compilation error:
  Database 'ANALYTICS_PRD' does not exist or not authorized.'
  
  I was wondering if that was a permissions issue as I do not see that db in Snowflake?

**Comment by britt-allen** [comment]
> I'm getting the error
> 
> 'Database Error in model imputation__detector_summary (models/marts/imputation/imputation__detector_summary.sql) 002003 (02000): SQL compilation error: Database 'ANALYTICS_PRD' does not exist or not authorized.'
> 
> I was wondering if that was a permissions issue as I do not see that db in Snowflake?

I'll take a look at your perms now

**Comment by ian-r-rose** [comment]
Just fixed it (I think)! We just landed our first incremental model in `ANALYTICS_PRD`, which meant that the cloning step needed to be able to read from that database

**Comment by tnrahim** [comment]
This will close #275

File: transform/models/intermediate/clearinghouse/_clearinghouse.yml [hunk 1]
```diff
@@ -1,54 +1,6 @@
 version: 2
 
 models:
-  - name: int_clearinghouse__station_meta
```
[the remark this answers is not in the document]
Comment (tnrahim) [reply transform/models/intermediate/clearinghouse/_clearinghouse.yml:4]: I've restored the documentation and updated the description.

File: transform/models/intermediate/vds/int_vds__active_stations.sql [hunk 2]
```diff
@@ -23,8 +23,7 @@ active_station as (
     inner join
         station_meta as sm
         on
-            dr.active_date >= sm._valid_from
-            and (dr.active_date < sm._valid_to or sm._valid_to is null)
+            {{ get_scd_2_data('dr.active_date','sm._valid_from','sm._valid_to') }}
```
[the remark this answers is not in the document]
Comment (tnrahim) [reply transform/models/intermediate/vds/int_vds__active_stations.sql:26]: Updated this in all the places I could spot.

File: transform/models/intermediate/diagnostics/int_diagnostics__detector_status.sql [hunk 3]
```diff
@@ -25,6 +26,23 @@ district_feed_check as (
...
+        dm.district,
+        cast(dm.physical_lanes as NUMBER(1, 0)) as lane,
+        sps.* exclude (sample_date, district, station_id, lane)
+    from {{ ref ('int_vds__active_stations') }} as dm
+    left outer join source as sps
+        on
+            dm.active_date = sps.sample_date
+            and dm.station_id = sps.station_id
+            and dm.district = sps.district
+            and dm.physical_lanes = sps.lane
```
[the remark this answers is not in the document]
Comment (tnrahim) [reply transform/models/intermediate/diagnostics/int_diagnostics__detector_status.sql:42]: This join is filling in the missing metadata from issue #226.
***
No gaps in dates, no null values
No gaps in stations, no null values
No gaps in detectors/lanes, no null values
No rows detectors/lanes that don't exist (i.e., the eighth lane in a highway that only has two lanes)
***
Without the logic there are many nulls present in the view 

***
select * from TRANSFORM_DEV.DBT_TRAHIM_DIAGNOSTICS.INT_DIAGNOSTICS__DETECTOR_STATUS  where station_id is null or lane is null or active_date is null or sample_date is null;
***

I didn't realize that physical lanes was was a count  of lanes. I'll update the code to use int_vds__detector_config instead in order ensure lane data is not missing.

File: transform/models/intermediate/diagnostics/int_diagnostics__detector_status.sql [hunk 4]
```diff
@@ -25,6 +26,23 @@ district_feed_check as (
     group by source.district
 ),
 
+source_with_detector_metadata as (
```
[the remark this answers is not in the document]
Comment (tnrahim) [reply transform/models/intermediate/diagnostics/int_diagnostics__detector_status.sql:29]: see below conversation

**Review by kengodleskidot — APPROVED** [review]
Great work @tnrahim!

**Pull request merged**